node/src/Pos/Wallet/Web/State/Transactions.hs

```diff
@@ -41,3 +48,27 @@ deleteWallet walId = do
   WS.removeWallet walId
   WS.removeTxMetas walId
   WS.removeHistoryCache walId
+
+-- | Apply some set of modifiers to a wallet.
+--   TODO Find out the significance of this set of modifiers and document.
+applyModifierToWallet
+  :: CId Wal
+  -> [CWAddressMeta] -- ^ Wallet addresses to add
+  -> [(WS.CustomAddressType, [(CId Addr, HeaderHash)])] -- ^ Custom addresses to add
+  -> UtxoModifier
+  -> [(CTxId, CTxMeta)] -- ^ Transaction metadata to add
+  -> Map TxId TxHistoryEntry -- ^ Entries for the history cache
+  -> [(TxId, PtxCondition)] -- ^ PTX Conditions
+  -> HeaderHash -- ^ New sync tip
+  -> Update ()
+applyModifierToWallet walId wAddrs custAddrs utxoMod
```

Comment:
I guess this one will require a bit more thought (but just because I guess it's a fiddly issue). Let me report an excerpt of a conversation I had with @pva701 a while ago:

> `CAccModifier` is outdated name of such thing which should be actually called *wallet modifier*. previously wallet was called account (so Acc in `CAccModifier` stands for "account"), so now we have this discrepancy. 
as you know, the wallet listens to blockchain, in sense that it reacts on each block (`BListener` class is provide interface for it).
each block may affect some wallet stored in wallet-db, namely, incoming and outgoing transactions may affect wallet-db's utxo, add and remove new addresses and so on and so forth.
so we consider each block from point of view of each stored wallet and try to find which things are affected for each of wallets.
so `CAccModifier` contains all these modifications for *one particular wallet*. for each block we compute `CAccModifier` corresponding to some wallet and then apply it to wallet-db.
yet another thing, that `CAccModifier` is monoid and hence `CAccModifier`s for several blocks (but corresponding to the same wallet) can be mappended and as result we get such `CAccModifier`, application of which gives the same wallet-db as for sequential applications of `CAccModifier`s for these blocks to wallet-db.
(you may see mappending in some places of code and it was done to reduce number of writing to wallet-db, especially when we sync with blockchain and iterate over whole blockchain)

@nc6 Is this excerpt of any help to you? 😀